File: dotnet/Allmark/Render/RenderChildren.cs
namespace Allmark.Render;

using Allmark.Types;

public static class RenderChildren
{
    public static void Execute(MarkdownNode node, RendererState state, bool? decode = true)
    {
        var children = node.Children;
        if (children != null && children.Count > 0)
        {
            foreach (var child in children)
            {
                RenderNode.Execute(child, state, decode);
            }
        }
    }
}

File: zig/src/render/renderChildren.zig
const std = @import("std");
const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const RendererState = @import("../types/RendererState.zig").RendererState;
const Renderer = @import("../types/Renderer.zig").Renderer;
const renderNode = @import("renderNode.zig").renderNode;

pub fn renderChildren(
    node: *const MarkdownNode,
    state: *RendererState,
    decode: bool,
) void {
    if (node.children) |children| {
        if (children.len > 0) {
            for (children) |child| {
                renderNode(child, state, decode);
            }
        }
    }
}
